odoo/addons/base/models/res_config.py

```diff
@@ -749,3 +749,36 @@ def get_config_warning(self, msg):
         if (action_id):
             return RedirectWarning(msg % values, action_id, _('Go to the configuration panel'))
         return UserError(msg % values)
+
+    @api.model
+    def create(self, values):
+        # Optimisation: saving a res.config.settings even without changing any
+        # values will trigger the write of all related values. This in turn may
+        # trigger chain of further recomputation. To avoid it, delete values
+        # that were not changed.
+        for field in self._fields.values():
+            # E.g.
+            # qr_code = fields.Boolean(related='company_id.qr_code', readonly=False)
+            if not (field.related and not field.readonly and field.name in values):
+                continue
+            new_value = values[field.name]
+            # E.g. field.related is ['company_id', 'qr_code']
+            fname0 = field.related[0]
+            if fname0 not in values:
+                continue
+            # e.g. company_id value (integer)
+            current_value = values[fname0]
+            # e.g. company_id value (browsed record)
+            current_value = self.env[self._fields[fname0].comodel_name].browse(current_value)
+            for fname in field.related[1:]:
+                current_value = getattr(current_value, fname)
```

Comment:
Never use `getattr(record, name)` to get the value of a field, use `record[name]` instead.  The latter checks that `name` is a field name, while the former does not.